Makefile

```diff
@@ -1,18 +1,33 @@
-.PHONY: dev clean e2e push
+.PHONY: clean dev-public-cluster update-image-on-deployment push-tester-image deploy-e2e run-e2e
 
-dev:
-	terraform --version
-	cd devenv && mkdir -p state && terraform init && terraform apply -auto-approve
 
-clean:
-	rm -rf devenv/state devenv/.terraform.lock.hcl devenv/.terraform devenv/terraform.tfstate devenv/terraform.tfstate.backup
+# keep separate for simultaneous public/private dev without need for resource recreation
+clean-public:
+	rm -rf devenv/state devenv/public_cluster_tf/.terraform.lock.hcl devenv/public_cluster_tf/.terraform devenv/public_cluster_tf/terraform.tfstate devenv/public_cluster_tf/terraform.tfstate.backup
 
-e2e:
-	go test -v --count=1 --tags=e2e ./e2e
+dev-public-cluster:
+	terraform --version
+	cd devenv && mkdir -p state && cd public_cluster_tf && terraform init && terraform apply -auto-approve
 
-push:
+# aka make push (formerly known as make push-private)
+update-image-on-deployment:
 	echo "$(shell cat devenv/state/registry.txt)/app-routing-operator:$(shell date +%s)" > devenv/state/operator-image-tag.txt
 	az acr login -n `cat devenv/state/registry.txt`
 	docker build -t `cat devenv/state/operator-image-tag.txt` .
 	docker push `cat devenv/state/operator-image-tag.txt`
-	kubectl set image -n kube-system --kubeconfig devenv/state/kubeconfig deployments/app-routing-operator operator=`cat devenv/state/operator-image-tag.txt`
+	cd devenv && ./scripts/push_image.sh
```

Comment:
should probably just rename to `update-image`. Short and concise. 

The script you're calling should be renamed as well. Right now it's confusing because you're calling the push image script inside update image even though there's also push-tester-image below.

Reply:
after discussion renaming to:
make push
make e2e

getting rid of make push tester image entirely!